server.py

```diff
@@ -0,0 +1,123 @@
+from sys import argv
+from os import path, sep
+
+from urllib.parse import urlparse
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from http.cookies import SimpleCookie
+from cgi import FieldStorage
+
+from jinja2 import Environment, FileSystemLoader
+
+
+HOST = '127.0.0.1'
+PORT = 8001
+
+if len(argv) > 1:
+    PORT = int(argv[1])
+
+
+class Handler(BaseHTTPRequestHandler):
+
+    def _get_auth_cookies(self):
+        cookies = SimpleCookie(self.headers.get('Cookie'))
+        is_exist = False
+
+        try:
+            if int(cookies['auth'].value) > 0:
+                is_exist = True
+        except KeyError:
+            print(f'There is no entity with \'auth\' key')
+
+        return is_exist
+
+    def _set_template(self, page, **kwargs):
+        file_loader = FileSystemLoader(sep.join((path.dirname(path.abspath(__file__)), 'templates')))
+        env = Environment(loader=file_loader)
+        template = env.get_template(page)
+        output = template.render(kwargs)
+        self.wfile.write(output.encode('UTF-8'))
+
+    def _set_auth_cookies(self, value):
+        cookies = SimpleCookie()
+
+        cookies['auth'] = value
+
+        if value == 0:
+            cookies['auth']['expires'] = 'Thu, 01 Jan 1970 00:00:00 GMT'
+
+        self.send_header('Set-Cookie', cookies.output(header='', sep=''))
+
+    def _charge_handler(self):
+        pass
+
+    def _error_handler(self):
+        pass
+
+    def _form_handler(self):
+        pass
+
+    def _log_in_handler(self):
+        self._set_auth_cookies(100)
+
+    def _log_out_handler(self):
+        self._set_auth_cookies(0)
+
+    HANDLERS = {
+        'charge.html': _charge_handler,
+        'error.html': _error_handler,
+        'form.html': _form_handler,
+        'logIn.html': _log_in_handler,
+        'logOut.html': _log_out_handler,
+    }
+
+    def _set_headers(self, page=None, content_type='text/html'):
+        response_code = 200
+
+        if page == 'error.html':
+            response_code = 404
+
+        self.send_response(response_code)
+        self.send_header('Content-type', content_type)
+
+        if page is not None:
+            self.HANDLERS[page](self)
+
+        self.end_headers()
+
+    def do_GET(self):
+        page = urlparse(self.path).path[1:] + '.html'
+
+        if 'favicon.ico' in self.path or page not in self.HANDLERS.keys():
+            page = 'error.html'
+
+        self._set_headers(page)
+
+        if self._get_auth_cookies():
+            auth = '<form action="/logOut"><input type="submit" value="Log Out"></form>'
+        else:
+            auth = '<form action="/logIn"><input type="submit" value="Log In"></form>'
+
+        self._set_template(page, auth=auth)
+
+    def do_POST(self):
+        self._set_headers()
+        page = urlparse(self.path).path[1:] + '.html'
```

Comment:
in fact you start here like in `do_GET` from a common page detection, and then on next lines you already handle the `charge` use case as if it's for sure the only use case. Which is true for the particular task, but not so much scalable in case you'll want to add more POST endpoints. 
No need to fix it now, just consider for the future.